src/caffe/net.cpp

```diff
@@ -46,6 +46,9 @@ void Net<Dtype>::Init(const NetParameter& in_param) {
   // the current NetState.
   NetParameter filtered_param;
   FilterNet(in_param, &filtered_param);
+  if (phase_ == TRAIN) {
+    caffe::P2PSync<Dtype>::divide_batch_size(&filtered_param);
+  }
```

Comment:
Why only divide for the `TRAIN` phase? Why not the `TEST` phase too?

This appears to be working so far - I'm just curious.